content/blog/tikv-3.0ga-release-notes.md

```diff
@@ -0,0 +1,55 @@
+---
+Today we're very proud to announce the general availability of TiKV 3.0! Before release, TiKV 3.0 underwent a rigorous testing, including an [official Jepsen test](https://www.pingcap.com/blog/tidb-passes-jepsen-test-for-snapshot-isolation-and-single-key-linearizability/) with TiDB.
+
+In this version, we tackled the problem of stability at massive scales. Whether it's spanning hundreds of nodes, or storing over a trillion key-value pairs, we've seen our users and contributors put TiKV to the test in serious, real world, production scenarios. With 3.0, we've taken our ideas and lessons to bring a host of features that can better support these growing demands.
+
+## Steady at scale
+
+In TiKV 3.0 we've improved our system by:
+
+* **Optimizing the Raft heartbeat mechanism.** TiKV now adjusts the heartbeat frequency according to region activity. This means you'll see less CPU time and network traffic from idle regions.
+
+* **Distributing Garbage Collection.** The introduction of a distributed garbage collector improves performance on large scale clusters dramatically, leading to better stability though more consistent performance.
+
+* **Pessimistic Locking.** It's now possible to ask TiKV to treat your transactions pessimistically. This means you can take exclusive ownership over a value for a duration. As Rust developers, we really like the idea of ownership!
+
+* **Expanding our coprocessor.** With lots of new functionality, improved APIs, vector operations, and better ways to work with data, our coprocessor continues to evolve, allowing for better, more efficient ways to work with your data.
+
+* **Enhancing operator friendliness.** Human or machine, we've empowered our operators to get more out of TiKV by unifying our log format, adding new features to `tikv-ctl`, adding even more in depth metrics, and serving HTTP based metrics. This makes TiKV easier to operate, inspect, and enjoy.
+
+* **Refining request types.** While TiKV previously had commands like `BatchGet`, 3.0 brings a new `BatchCommands` request type. We've made taught TiKV how to handle batches of requests of differing kinds, leading to less data on the wire and better performance. We also added support for raw reversed scanning, and `Insert` semantics on prewrite.
+
+* **Reducing write amplication.** We were inspired by the great ideas from [WiscKey](https://www.usenix.org/system/files/conference/fast16/fast16-papers-lu.pdf) and implemented a feature we dubbed 'Titan'. It works best with larger (>1 KB) values.
+
+You can see all the changes in detail [here](https://github.com/tikv/tikv/blob/release-3.0/CHANGELOG.md).
+
+## Improved Performance
+
+Using [`go-ycsb`](https://github.com/pingcap/go-ycsb) we benchmarked TiKV 3.0.0-rc.2 against TiKV 2.1.12. We benchmarked a cluster of 3 TiKV nodes and 1 PD node. They were running on public cloud machines with 16 cores, 32 GB of RAM, a 400 GB NVMe SSD, and a 10Gbps network. We used the default settings of TiKV and PD for this.
```

Comment:
Should users concern why we used 3.0.0-rc.2 for benchmarking in the 3.0 GA announcement?

Reply:
I'm not sure, what do you think @breeswish ?

Reply:
please use 3.0-GA for benchmarking again.

Reply:
> please use 3.0-GA for benchmarking again.
Please could you assign this @siddontang

Reply:
+1 for we should use 3.0-GA. The benchmark result should be pretty close though.

Maybe @sticnarf can help you do this as well?

Reply:
I'll work on this :)